File: Python/leet_code/Easy/217. Contains Duplicate.py
# ===========Contains Duplicate================

# ? Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

# Example 1:
# Input: nums = [1,2,3,1]
# Output: true

# Example 2:
# Input: nums = [1,2,3,4]
# Output: false

def containsDuplicate(nums):
    return len(set(nums)) != len(nums)
    

# Runtime:  99.89% | Memory: 16.55%

File: Rust/LeetCode/Easy/217. Contains Duplicate.rs
// ===========Contains Duplicate================

// ? Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

// Example:
// Input: nums = [1,2,3,1]
// Output: true

use std::collections::HashSet;

pub fn contains_duplicate(nums: Vec<i32>) -> bool {
    let nums_set: HashSet<i32> = nums.iter().cloned().collect();
    nums.len() != nums_set.len()
}

// Runtime:  84.08% | Memory: 70.31%